sdk/python/kfp/dsl/graph_component.py

```diff
@@ -86,6 +92,106 @@ def __init__(
         self.component_spec.implementation.graph = pipeline_spec
         self.component_spec.platform_spec = platform_spec
 
+    def _detect_workspace_usage_in_tasks(
+            self, tasks: List[pipeline_task.PipelineTask]) -> bool:
+        """Detects if any task in the list uses workspace features.
+
+        Args:
+            tasks: List of pipeline tasks to check for workspace usage.
+
+        Returns:
+            True if any task uses workspace features, False otherwise.
+        """
+
+        def _check_value_for_workspace_placeholder(value: Any) -> bool:
+            """Recursively check a value for workspace placeholder usage."""
+            if isinstance(value, str):
+                return dsl.WORKSPACE_PATH_PLACEHOLDER in value
+            elif isinstance(value, dict):
+                return any(
+                    _check_value_for_workspace_placeholder(v)
+                    for v in value.values())
+            elif isinstance(value, list):
+                return any(
+                    _check_value_for_workspace_placeholder(item)
+                    for item in value)
+            return False
+
+        for task in tasks:
+            if hasattr(task, 'inputs') and task.inputs:
+                for input_value in task.inputs.values():
+                    if _check_value_for_workspace_placeholder(input_value):
+                        return True
+
+            if hasattr(task, 'args') and task.args:
+                if _check_value_for_workspace_placeholder(task.args):
+                    return True
+
+            if hasattr(task, 'container_spec') and task.container_spec:
+                if hasattr(task.container_spec,
+                           'args') and task.container_spec.args:
+                    if _check_value_for_workspace_placeholder(
+                            task.container_spec.args):
+                        return True
+                if hasattr(task.container_spec,
+                           'command') and task.container_spec.command:
+                    if _check_value_for_workspace_placeholder(
+                            task.container_spec.command):
+                        return True
```

Comment:
I think based on `needsWorkspaceMount` in `backend/src/v2/driver/driver.go`, it only looks at input parameters and since `task.inputs` would show something like this `{'workspace_path': '{{$.workspace_path}}'}`, we can just use `task.inputs`.